File: frontend/web/src/api/messages/getMessages.ts
import axiosInstance from "../../utils/AxiosInstance.ts"

const getMessages = async (chatId: string | undefined) => {
    const token = localStorage.getItem("token");

    const { data } = await axiosInstance.get(`/api/messages/msg?chatId=${chatId}`, {
        headers: {
            Authorization: `Bearer ${token}`
        }
    });

    return data.messages;
}

export default getMessages;

File: frontend/desktop/python-client/PyQt5App/api/messages/getMessages.py
import requests
from PyQt5App.utils.Address import address
from PyQt5App.utils.sessionManager import SessionManager

class GetMessages:
    def __init__(self, chat_id):
        self.address = address()
        self.sessionManager = SessionManager()
        self.token = self.sessionManager.get_token()
        self.chat_id = chat_id

    def getMessages(self):
        response = requests.get(
            f"{self.address}/api/messages/{self.chat_id}",
            headers={
                "Authorization": f"{self.token}",
            }
        )
        data = response.json()
        return data.get("messages", [])